src/Components/Question.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from "react";
+import Options from "./Options";
+import { useState } from "react";
+import { Result } from "./Result";
+import "../tailwind.css";
+import { useParams } from "react-router-dom";
+import { useQuiz } from "../Context/QuizProvider";
+
+export default function QuestionComponent() {
+  const { state, dispatch } = useQuiz();
+  const { quizId } = useParams();
+  const [showResult, setShowResult] = useState(false);
+  const question_no = state.currQues;
+  const ques = state.quizzes[Number(quizId) - 1].questions[state.currQues - 1];
+  const nextQuestionHandler = () => {
+    if (state.currQues === 5) {
+      setShowResult(!showResult);
+    } else {
+      dispatch({ type: "NEXT_QUESTION" });
+    }
+  };
+  return (
+    <div>
+      {!showResult ? (
```

Comment:
Extract this into multiple components as the logic is not clear with all these conditionals.